python/ml4ir/applications/ranking/tests/test_lr_schedule.py

```diff
@@ -216,8 +225,88 @@ def test_cyclic_lr_in_training_pipeline(self):
 
         assert np.all(np.isclose(lr_gold, lr_list))
 
+    def test_reduce_lr_on_plateau_in_training_pipeline(self):
+        """Test reduce lr on plateau"""
+        self.model_config_file = MODEL_CONFIG_REDUCE_LR_ON_PLATEAU
+        Logger = logging_utils.setup_logging(
+            reset=True,
+            file_name=os.path.join(INPUT_DIR + 'ranklib', "output_log.csv"),
+            log_to_file=True,
+        )
+
+        io = LocalIO()
+        feature_config = self.parse_config(TFRecordTypeKey.SEQUENCE_EXAMPLE, self.feature_config_yaml_convert_to_clicks,
+                                           io)
+        model_config = io.read_yaml(self.model_config_file)
+
+        dataset = RelevanceDataset(
+            data_dir=INPUT_DIR + '/ranklib',
+            data_format=DataFormatKey.RANKLIB,
+            feature_config=feature_config,
+            tfrecord_type=TFRecordTypeKey.SEQUENCE_EXAMPLE,
+            batch_size=32,
+            file_io=io,
+            preprocessing_keys_to_fns={},
+            logger=Logger,
+            keep_additional_info=KEEP_ADDITIONAL_INFO,
+            non_zero_features_only=NON_ZERO_FEATURES_ONLY,
+            max_sequence_size=319,
+        )
+
+        # Define interaction model
+        interaction_model: InteractionModel = UnivariateInteractionModel(
+            feature_config=feature_config,
+            feature_layer_keys_to_fns={},
+            tfrecord_type=TFRecordTypeKey.SEQUENCE_EXAMPLE,
+            max_sequence_size=319,
+            file_io=io,
+        )
+
+        # Define loss object from loss key
+        loss: RelevanceLossBase = loss_factory.get_loss(
+            loss_key=LossKey.RANK_ONE_LISTNET, scoring_type=ScoringTypeKey.POINTWISE
+        )
+
+        # Define scorer
+        scorer: ScorerBase = RelevanceScorer.from_model_config_file(
+            model_config_file=self.model_config_file,
+            interaction_model=interaction_model,
+            loss=loss,
+            logger=Logger,
+            file_io=io,
+        )
+
+        optimizer: Optimizer = get_optimizer(model_config=model_config)
+
+        # Combine the above to define a RelevanceModel
+        relevance_model: RelevanceModel = RankingModel(
+            feature_config=feature_config,
+            tfrecord_type=TFRecordTypeKey.SEQUENCE_EXAMPLE,
+            scorer=scorer,
+            optimizer=optimizer,
+            model_file=None,
+            file_io=io,
+            logger=Logger,
+            model_config=model_config,
+        )
```

Comment:
@mohazahran Can you test this without creating a RankingModel? I ask this because creating new models is resulting in memory overflow on the autobuilds.

Reply:
One (admittedly convoluted) way to do this is by just calling `on_epoch_end()` multiple times to test the learning rate updates. I agree that this is essentially testing `ReduceLrOnPlateau` and not whether it works in ml4ir. So for now, you can leave it as is I think. 

I'm also wondering if we even need an expensive test such as this one from our end. Maybe the real test should be to see if the right monitoring metric was added by the `add_scheduler_as_callback` and everything was configured right. That's the ml4ir part really, right?

https://github.com/keras-team/keras/blob/v2.7.0/keras/callbacks.py#L2603-L2724

Reply:
@mohazahran I hate to be a party pooper here but these comments that are left unanswered (perhpas you synced offline) is not reassuring. 
There are two issues with tests in ml4ir: 
- We run out of memory
- They take too long
The tests here contribute to both problems. Can we do something else to validate the functionality?
If not:
- Just use 3 epochs
- Ensure you are using toy data like 20 examples and not 2K 
Thoughts?

Reply:
We have not synced offline. Even I am awaiting replies to move forward with the PR.

Reply:
@balikasg I don't think the data size really matters other than adding a few (milli) seconds to the training time. The fact that we create yet another model is the real issue and should be looked at in more detail (not part of this PR)